daemon/logger/splunk/splunk.go

```diff
@@ -22,29 +25,66 @@ import (
 )
 
 const (
-	driverName                  = "splunk"
-	splunkURLKey                = "splunk-url"
-	splunkTokenKey              = "splunk-token"
-	splunkSourceKey             = "splunk-source"
-	splunkSourceTypeKey         = "splunk-sourcetype"
-	splunkIndexKey              = "splunk-index"
-	splunkCAPathKey             = "splunk-capath"
-	splunkCANameKey             = "splunk-caname"
-	splunkInsecureSkipVerifyKey = "splunk-insecureskipverify"
-	splunkFormatKey             = "splunk-format"
-	splunkVerifyConnectionKey   = "splunk-verify-connection"
-	envKey                      = "env"
-	labelsKey                   = "labels"
-	tagKey                      = "tag"
+	driverName                     = "splunk"
+	splunkURLKey                   = "splunk-url"
+	splunkTokenKey                 = "splunk-token"
+	splunkSourceKey                = "splunk-source"
+	splunkSourceTypeKey            = "splunk-sourcetype"
+	splunkIndexKey                 = "splunk-index"
+	splunkCAPathKey                = "splunk-capath"
+	splunkCANameKey                = "splunk-caname"
+	splunkInsecureSkipVerifyKey    = "splunk-insecureskipverify"
+	splunkFormatKey                = "splunk-format"
+	splunkVerifyConnectionKey      = "splunk-verify-connection"
+	splunkGzipCompressionKey       = "splunk-gzip"
+	splunkGzipCompressionLevelKey  = "splunk-gzip-level"
+	envKey                         = "env"
+	labelsKey                      = "labels"
+	tagKey                         = "tag"
 )
 
+const (
+	defaultPostMessagesFrequency       = 5 * time.Second
+	defaultPostMessagesBatchSize       = 1000
+	defaultBufferMaximum               = 10 * defaultPostMessagesBatchSize
+	defaultStreamChannelSize           = 4 * defaultPostMessagesBatchSize
+)
+
+const (
+	envVarPostMessagesFrequency = "SPLUNK_LOGGING_DRIVER_POST_MESSAGES_FREQUENCY"
+	envVarPostMessagesBatchSize = "SPLUNK_LOGGING_DRIVER_POST_MESSAGES_BATCH_SIZE"
+	envVarBufferMaximum         = "SPLUNK_LOGGING_DRIVER_BUFFER_MAX"
+	envVarStreamChannelSize     = "SPLUNK_LOGGING_DRIVER_CHANNEL_SIZE"
+)
+
+type splunkLoggerInterface interface {
+	logger.Logger
+	worker()
+}
+
 type splunkLogger struct {
 	client    *http.Client
 	transport *http.Transport
 
 	url         string
 	auth        string
 	nullMessage *splunkMessage
+
+	// http compression
+	gzipCompression bool
+	gzipCompressionLevel int
+
+	// Advanced options
+	postMessagesFrequency time.Duration
+	postMessagesBatchSize int
+	bufferMaximum int
+	streamChannelSize int
+
+	// For synchronization between background worker and logger
+	stream     chan *splunkMessage
```

Comment:
It's not 100% clear how these locks/condition variables work with each other and the goroutine. I think we need to explain it.